lib/Util.ts

```diff
@@ -75,4 +76,39 @@ function buildParamsStringifyOptions(queryParams: IRequestQueryParams): any  {
     }
 
     return options;
-}
\ No newline at end of file
+}
+
+/**
+ * Decompress/Decode gzip encoded JSON
+ * Using Node.js built-in zlib module
+ *
+ * @param {Buffer} buffer
+ * @param {string} charset? - optional; defaults to 'utf-8'
+ * @return {Promise<string>}
+ */
+export async function decompressGzippedContent(buffer: Buffer, charset?: string): Promise<string> {
+    return new Promise<string>(async (resolve, reject) => {
+        zlib.gunzip(buffer, function (error, buffer) {
+            if (error) {
+                reject(error);
+            }
+
+            resolve(buffer.toString(charset || 'utf-8'));
+        });
+    })
+}
+
+/**
+ * Obtain Response's Content Charset.
+ *
+ * @param {IHttpClientResponse} response
+ * @return {string} - Content Encoding Charset; Default=utf-8
+ */
+export function obtainContentCharset (response: IHttpClientResponse) : string {
+  // Find the charset, if specified.
+  // Match with regex "charset="
+  const contentType: string = response.message.headers['content-type'] || '';
+  const matches: (RegExpMatchArray|null) = contentType.match(/charset=([^;,\r\n]+)/i);
```

Comment:
Could you maybe clarify a little bit what this regex is doing? It looks fine to me, but just for readability it might be good.

Maybe just add an example (e.g. `FOOcharset=BAR` will get matched to `[charset=BAR, BAR]`) - that also makes it clear why you're taking matches[1] below.

Reply:
Yeah, Gotcha! Will add helper comments to clarify the functionality, and add an example of matching for the charset 👍